### PR TITLE
Update 3rd-party plugins to NIM compatible versions

### LINKED ISSUE #41: Update MathType and WProofreader demos with NIM compatible versions
Follow-up of #40.

2 demos with mentioned plugins have those commented-out since there are no NIM compatible versions yet. After new plugins versions are released, demos should be updated.

WProofreader was already tested with `alpha` version as mentioned here - https://github.com/ckeditor/ckeditor5-demos/pull/40#issuecomment-2172881625.

### PR DESCRIPTION
**Pull request by f1ames**
This PR updates MathType and WProofreader to latest versions (NIM compatible).

I also fixed license key message as it mentioned `index.ts` instead of `index.js`.

Fixes #41.

### CODE REVIEW

**Review by pszczesniak — APPROVED** [review]
LGTM :+1:

**Comment by Witoso** [comment]
Related to this #43 

In some older snippets, I saw a comment above this plugin with it being commented out. Maybe let's do this?

**Comment by f1ames** [comment]
> Related to this #43 
> 
> In some older snippets, I saw a comment above this plugin with it being commented out. Maybe let's do this?

@Witoso You mean to comment out WProofreader plugin by default? Or addin explanation on "blocked" localhost? For commenting out this plugin, it's "WProofreader" so I would assume it should be there, enabled by default 🤔

**Comment by f1ames** [comment]
Ok, I have added comment to each demo with WProofreader to address #43. Also fixed some demos wich weren't updated before.

@pszczesniak could you take another look if the message makes sense?

File: feature-rich/index.js [hunk 1]
```diff
@@ -22,6 +22,9 @@ const CKBOX_TOKEN_URL = '';
 
 // WProofreader plugin require a license key to work properly.
 // For more info how to get the key, see https://ckeditor.com/docs/ckeditor5/latest/features/spelling-and-grammar-checking.html.
+//
+// IMPORTANT: WProofreader has a mechanism preventing it to run on 'localhost' domain. To test it locally, you need to use
+// a different domain name, by, for example, mapping 'localhost' to 'fake.ckeditor.com' in your '/etc/hosts' file.
```
Comment (Witoso) [inline feature-rich/index.js:27]:
```suggestion
// a different domain name, by, for example, mapping 'localhost' to 'localhost.dev' in your '/etc/hosts' file.
```

I would rather not leak fake.ckeditor.com.

**Review by Witoso — CHANGES_REQUESTED** [review]

File: feature-rich/index.js (hunk 1 above)
Comment (pszczesniak) [reply feature-rich/index.js:27]: `.dev` domains can be locally? AFAIR when Google bought this makes some redirects even locally on Chrome? :thinking:

Maybe `dev.your-domain.com`?

Comment (Witoso) [reply feature-rich/index.js:27]: ah, you're right, I saw also `dev.localhost` pattern I think.

Comment (f1ames) [reply feature-rich/index.js:27]: Ok, I'll fix it 👍

I'm just confirming now how it works exactly. I see with trial key, all domains are allowed by default 🤔 Probably, that's not the case for paid plan. I'm checking this and will adjust the message if needed.

Comment (f1ames) [reply feature-rich/index.js:27]: Ok, I talked to @Stalq, and it seems this is no longer relevant. It worked like that some time ago but now pricing model is based on the number of spellchecked words (AFAIU).

There is a way to change allowed/excluded domains on WSC side to e.g. to exclude some, but not automatically (requires contacting their support) so this is a really custom thing.

I'll restore the original comments. Sorry for not checking this in more details earlier 😓 🙈

**Comment by f1ames** [comment]
Merging now, since the only requested change is no longer relevant (due to https://github.com/ckeditor/ckeditor5-demos/pull/42#discussion_r1656960506).

**Pull request merged**